src/main/java/fiji/plugin/trackmate/action/CloseGapsByLinearInterpolationAction.java

```diff
@@ -0,0 +1,157 @@
+package fiji.plugin.trackmate.action;
+
+import fiji.plugin.trackmate.Model;
+import fiji.plugin.trackmate.Spot;
+import fiji.plugin.trackmate.SpotCollection;
+import fiji.plugin.trackmate.TrackMate;
+import fiji.plugin.trackmate.gui.TrackMateGUIController;
+import fiji.plugin.trackmate.gui.TrackMateWizard;
+import ij.gui.WaitForUserDialog;
+import net.imglib2.RealPoint;
+import org.scijava.plugin.Plugin;
+
+import javax.swing.*;
+import java.util.Set;
+
+/**
+ * This action allows to close gaps in tracks by creating new intermediate spots which are located at interpolated positions.
+ * This is useful if you want to measure signal intensity changing during time, even if the spot is not visible. Thus, trackmate
+ * is utilisable for Fluorescence Recovery after Photobleaching (FRAP) analysis.
+ *
+ * Author: Robert Haase, Scientific Computing Facility, MPI-CBG, rhaase@mpi-cbg.de
+ *
+ * Date: June 2016
+ *
+ */
+public class CloseGapsByLinearInterpolationAction extends AbstractTMAction {
+
+
+    public static final ImageIcon ICON = new ImageIcon(TrackMateWizard.class.getResource("images/spot_icon.png"));
+    public static final String NAME = "Close gaps by introducing new spots";
+
+    public static final String KEY = "CLOSE_GAPS_BY_LINEAR_INPERPOLATION";
+    public static final String INFO_TEXT = "<html>" +
+            "This action closes gaps in tracks by introducing new spots. The spots positions and size are calculated using linear interpolation." +
+            "</html>";
+
+    @Override
+    public void execute(final TrackMate trackmate) {
+        final Model model = trackmate.getModel();
+
+
+        Set<Integer> trackIDs = model.getTrackModel().trackIDs(true);
+
+        // go through all tracks
+        for (Integer trackID : trackIDs)
+        {
+            Set<Spot> spots = model.getTrackModel().trackSpots(trackID);
```

Comment:
Use TrackModel.getDepthFirstIterator instead.
Then test the action on a track that divides.